nasa-gibs/worldview
# Vector Orbit Tracks

## Description

Replace WMS orbit tracks with vector orbit tracks

Fixes #2279 

- [x] Fix issue where vector styles are base off of default mapbox proj and use current proj and update all styles accordingly 
- [x] Customize `ol-mapbox-style` library to avoid cut-off labels by tile edges
- [x] Update Orbit track styles for each label
- [x] Update Orbit track layer configs  
- [x] Make the vector dialog follow orbit tracks features as you click on them (`modalShouldFollowClicks` in layer config)
- [x] Prevent from being able to click on orbit track line (can only click points and text) (`clickDisabledFeatures` in layer config)
- [x] Use package-patch plugin instead of forked `ol-mapbox-style` lib
- [x] fix hls bug
- [x] make wms/vector split
- [x] make labels larger
[If this is a relatively large or complex change, start a discussion by explaining why you chose the solution you did and what alternatives you considered, etc...]

## How To Test

[Provide whatever information a reviewer might need to know in order to verify that the changes made in this PR are working as expected. If there are special build steps that need to be taken in order to get these changes to run (building while on a separate branch, running `npm ci`, etc) include them here.]

- For bugfixes: What steps need to be taken in the UI to verify the bug is fixed?
- For enhancements and features: What is the expected functionality being added/modified? How can a reviewer verify this?

1. Make sure styles for vector layers look good (especially orbit tracks)
2.  Make sure vector layer interactions work well (especially orbit tracks)
3. Check that vector layers work with WVS

## PR Submission Checklist

This is simply a reminder of what we are going to look for before merging your code.

- I have read the [CONTRIBUTING](https://github.com/nasa-gibs/worldview/blob/main/.github/CONTRIBUTING.md) doc
- I have added necessary documentation (if applicable)
- I have added tests that prove my fix is effective or that my feature works (if applicable)
- Any dependent changes have been merged and published in downstream modules (if applicable)

@nasa-gibs/worldview
